### PR TITLE
Update typescript to 4.4.2

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
## What does this pull request do?

This replaces the Dependabot pull request #743. It updates TypeScript to 4.4.2. This involves changes to the code, to add type information to caught values. See commit message for more information.

## What is the intent behind these changes?

To keep our dependencies up to date.

### CODE REVIEW

**Comment by lawrence-forooghian** [comment]
Putting this back to draft, realised it's a bit rough around the edges.

**Review by Gweaton — APPROVED** [review]
Nice approach, looks good (just need to sort the merge conflicts)

File: server/authentication/passport.ts [hunk 1]
```diff
@@ -66,6 +66,12 @@ export default function passportSetup(app: Application, hmppsAuthService: HmppsA
 
           verified(null, loggedInUser)
         } catch (error) {
+          // It almost certainly is an Error, but we need to convince the compiler
```
Comment (tomsmyers) [inline server/authentication/passport.ts:69]: what actually happens in the framework if we don't call the verified callback with the error?

could it be better to just cast the error to an `Error` and pass it anyway?

**Review by tomsmyers — APPROVED** [review]
lgtm - just one question about the passport verified callback

File: server/authentication/passport.ts (hunk 1 above)
Comment (lawrence-forooghian) [reply server/authentication/passport.ts:69]: Or we could create our own `Error` explaining what's gone wrong and the non-`Error` value we caught?

Comment (lawrence-forooghian) [reply server/authentication/passport.ts:69]: Hang on, wait, that's exactly what I've done. I reckon this approach is alright - it's unlikely to occur and we'll still describe the value we caught.

**Pull request merged**